src/omnicore/rpctxobject.cpp

```diff
@@ -256,6 +256,7 @@ void populateRPCTypeTradeOffer(CMPTransaction& omniObj, Object& txobj)
 
     // Check levelDB to see if the amount for sale has been amended due to a partial purchase
     // TODO: DEx phase 1 really needs an overhaul to work like MetaDEx with original amounts for sale and amounts remaining etc
+    // TODO: if the amount of the transaction is not used here, then the BTC amount should be recalculated (?)
```

Comment:
Sure:

Alice has only 50 MSC, but offers 1000 MSC for 1000 BTC. The (aweful) way how it's currently handled: the transaction is valid, and [the offer is adjusted](https://github.com/dexX7/bitcoin/blob/d761109327539abfd44af5e050f091ae32662db8/src/omnicore/dex.cpp#L131-L147) to 50 MSC for 50 BTC. The adjusted MSC amount is [fetched from the DB](https://github.com/dexX7/bitcoin/blob/0356f8f8399b6feac7d5e0d2fe7d62ee56016972/src/omnicore/rpctxobject.cpp#L264-L265), but the BTC amount isn't updated:

```java
omni_getbalance moR6aNzFHKSMUdqKb3vDCdetM7pLBHozKY 1
{
  "balance" : "50.00000000",  // <----
  "reserved" : "0.00000000"
}
```
```java
// DEx Offer of 1000 MSC for 1000 BTC
omni_sendrawtx "moR6aNzFHKSMUdqKb3vDCdetM7pLBHozKY" "0001001400000001000000174876e800000000174876e8000a000000000098968001"
f5c71761e1f6588784720bebf1eecbe3113b99a17d4857ec776352af566293ec
```
```java
setgenerate true 1
[
  "5a52c577e14918128425d182174cd6b36eaa17d1b92b68604145c5874476cedd"
]
```
```java
omni_gettransaction f5c71761e1f6588784720bebf1eecbe3113b99a17d4857ec776352af566293ec
{
  "txid": "f5c71761e1f6588784720bebf1eecbe3113b99a17d4857ec776352af566293ec",
  "fee": "0.00010000",
  "sendingaddress": "moR6aNzFHKSMUdqKb3vDCdetM7pLBHozKY",
  "ismine": true,
  "version": 1,
  "type_int": 20,
  "type": "DEx Sell Offer",
  "propertyid": 1,
  "divisible": true,
  "amount": "50.00000000",  // <----
  "bitcoindesired": "1000.00000000",  // <---- !!
  "timelimit": 10,
  "feerequired": "0.10000000",
  "action": "new",
  "valid": true,
  "blockhash": "5a52c577e14918128425d182174cd6b36eaa17d1b92b68604145c5874476cedd",
  "blocktime": 1439467162,
  "block": 103,
  "confirmations": 1
}
```